torch/csrc/jit/export.cpp

```diff
@@ -550,6 +553,24 @@ class ScriptModuleSerializer {
   }
 
  private:
+  void writeMaterializedInterfaceInstances(const script::Module& module) {
```

Comment:
This approach creates a second pass over the object hierarchy that can easily get out of sync with the pickler. It already is out of sync: what about tuples? what about nested modules? The pickler should be modified to keep a list of all the types it wrote, and that should be added to the list of types that writeCode writes.